types/transformer/artifacts/java.go

```diff
@@ -62,28 +62,15 @@ type JavaPackaging string
 
 // JarArtifactConfig defines a JarArtifactConfig struct
 type JarArtifactConfig struct {
-	DeploymentFile                    string            `yaml:"deploymentFile"`
-	JavaVersion                       string            `yaml:"javaVersion"`
-	BuildContainerName                string            `yaml:"buildContainerName"`
-	DeploymentFileDirInBuildContainer string            `yaml:"deploymentFileDirInBuildContainer"`
-	EnvVariables                      map[string]string `yaml:"envVariables"`
-	Port                              int32             `yaml:"port"`
+	Port               int32             `yaml:"port"`
+	JavaVersion        string            `yaml:"javaVersion"`
+	BuildContainerName string            `yaml:"buildContainerName"`
+	DeploymentFilePath string            `yaml:"deploymentFilePath"`
+	EnvVariables       map[string]string `yaml:"envVariables"`
 }
 
 // WarArtifactConfig defines a WarArtifactConfig struct
-type WarArtifactConfig struct {
-	DeploymentFile                    string            `yaml:"deploymentFile"`
-	JavaVersion                       string            `yaml:"javaVersion"`
-	BuildContainerName                string            `yaml:"buildContainerName"`
-	DeploymentFileDirInBuildContainer string            `yaml:"deploymentFileDirInBuildContainer"`
-	EnvVariables                      map[string]string `yaml:"envVariables"`
-}
+type WarArtifactConfig JarArtifactConfig
 
 // EarArtifactConfig defines a EarArtifactConfig struct
-type EarArtifactConfig struct {
-	DeploymentFile                    string            `yaml:"deploymentFile"`
-	JavaVersion                       string            `yaml:"javaVersion"`
-	BuildContainerName                string            `yaml:"buildContainerName"`
-	DeploymentFileDirInBuildContainer string            `yaml:"deploymentFileDirInBuildContainer"`
-	EnvVariables                      map[string]string `yaml:"envVariables"`
-}
+type EarArtifactConfig JarArtifactConfig
```

Comment:
Port does not have any meaning in War. If we want to reuse, we can see if promoted types work, and have port alone listed directly in Jar.

Reply:
It should be possible to set the port even in `war` case. Spring Boot allows it during development.

Reply:
it won't be possible, since war gets deployed to a running web server, and the web server has a port set. Spring boot during development  might be starting an external webserver in the port, and just puts the war (which does not have the port info) in that server.